src/__tests__/render.test.tsx

```diff
@@ -285,3 +287,133 @@ test('RenderAPI type', () => {
   render(<Banana />) as RenderAPI;
   expect(true).toBeTruthy();
 });
+
+test('should throw when rendering a string outside a text component', () => {
+  expect(() =>
+    render(<View>hello</View>, {
+      experimentalValidateStringsRenderedInText: true,
+    })
+  ).toThrowError('Text strings must be rendered within a host Text component.');
+});
+
+test('should throw an error when rerendering with text outside of Text component', () => {
+  // eslint-disable-next-line no-console
+  console.error = (errorMessage: string) => {
+    if (
+      !errorMessage.includes(
+        'The above error occurred in the <Profiler> component'
+      )
+    ) {
+      originalConsoleError(errorMessage);
+    }
+  };
+  const { rerender } = render(<View />, {
+    experimentalValidateStringsRenderedInText: true,
+  });
+
+  expect(() => rerender(<View>hello</View>)).toThrowError(
+    'Text strings must be rendered within a host Text component.'
+  );
+  // eslint-disable-next-line no-console
+  console.error = originalConsoleError;
+});
+
+const ErrorComponent = () => {
+  const [shouldDisplayText, setShouldDisplayText] = React.useState(false);
+
+  if (!shouldDisplayText) {
+    return (
+      <TouchableOpacity
```

Comment:
Pls use `<Pressable>` in tests, as recommended future-proof alternative to `TouchableXxx`